collab-folder/src/core/view.rs

```diff
@@ -362,12 +359,16 @@ pub(crate) fn view_from_map_ref<T: ReadTxn>(
     children,
     created_at,
     layout,
-    icon_url,
-    cover_url,
+    icon,
     is_favorite,
   })
 }
 
+pub fn icon_from_view_map<T: ReadTxn>(map_ref: &MapRef, txn: &T) -> Option<ViewIcon> {
+  let icon_str = map_ref.get_str_with_txn(txn, VIEW_ICON)?;
+  serde_json::from_str::<ViewIcon>(&icon_str).ok()
+}
```

Comment:
There're two suggestions.

1. still keep this function, remove the `pub` keyword and rename it to `get_icon_from_view_map`.
2. remove this function, and replace the above caller with this new code.

```rs
  let icon = map_ref
    .get_str_with_txn(txn, VIEW_ICON)
    .and_then(|icon_str| serde_json::from_str::<ViewIcon>(&icon_str).ok());
```